presto-native-execution/presto_cpp/main/operators/tests/UnsaferowShuffleTest.cpp

```diff
@@ -356,12 +368,33 @@ TEST_F(UnsafeRowShuffleTest, endToEnd) {
   size_t numPartitions = 5;
   size_t numMapDrivers = 2;
 
-  TestShuffle shuffle(pool(), numPartitions, 1 << 20 /* 1MB */);
+  auto shuffle =
+      std::make_shared<TestShuffle>(pool(), numPartitions, 1 << 20 /* 1MB */);
   auto data = vectorMaker_.rowVector({
       makeFlatVector<int32_t>({1, 2, 3, 4, 5, 6}),
       makeFlatVector<int64_t>({10, 20, 30, 40, 50, 60}),
   });
-  runShuffleTest(&shuffle, numPartitions, numMapDrivers, {data});
+  runShuffleTest(shuffle, numPartitions, numMapDrivers, {data});
+}
+
+TEST_F(UnsafeRowShuffleTest, persistentShuffleDeser) {
+  std::string serializedReadInfo = "{\n"
+      "  \"rootPath\": \"abc\",\n"
+      "  \"numPartitions\": 11\n"
+      "}";
+  LocalShuffleReadInfo readInfo =
+      LocalShuffleInfoDeserializer::deserializeReadInfo(serializedReadInfo);
+  EXPECT_EQ(readInfo.rootPath, "abc");
+  EXPECT_EQ(readInfo.numPartitions, 11);
+
+  std::string serializedWriteInfo = "{\n"
+      "  \"rootPath\": \"efg\",\n"
+      "  \"numPartitions\": 12\n"
+      "}";
+  LocalShuffleWriteInfo writeInfo =
+      LocalShuffleInfoDeserializer::deserializeWriteInfo(serializedWriteInfo);
+  EXPECT_EQ(writeInfo.rootPath, "efg");
+  EXPECT_EQ(writeInfo.numPartitions, 12);
```

Comment:
Shall we add a test cases that (1) pass bad serialized string to decode; (2) bad type?